deeprankcore/trainer.py

```diff
@@ -459,12 +477,15 @@ def set_loss(self):
                 weight=self.weights, reduction="mean")
 
 
-    def train( # pylint: disable=too-many-arguments
+    def train( # pylint: disable=too-many-arguments, too-many-branches, too-many-locals
         self,
         nepoch: int = 1,
+        batch_size: int = 32,
+        shuffle: bool = True,
         earlystop_patience: Optional[int] = None,
         earlystop_maxgap: Optional[float] = None,
         validate: bool = False,
+        num_workers: int = 0,
```

Comment:
Should this be an `Optional`?

Reply:
No, [Optional type](https://www.infoworld.com/article/3630372/get-started-with-python-type-hints.html#toc-5) means that they can either be set to the given type or to `None`.

This has been used inconsistently in the code until now, but will be cleaned up in one of the issues we have.